src/sys/socket/addr.rs

```diff
@@ -1035,28 +832,10 @@ pub mod sys_control {
 
     ioctl_readwrite!(ctl_info, CTL_IOC_MAGIC, CTL_IOC_INFO, ctl_ioc_info);
 
-    #[derive(Copy, Clone)]
     #[repr(C)]
+    #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
     pub struct SysControlAddr(pub libc::sockaddr_ctl);
 
-    impl PartialEq for SysControlAddr {
-        fn eq(&self, other: &Self) -> bool {
-            let (inner, other) = (self.0, other.0);
-            (inner.sc_id, inner.sc_unit) ==
```

Comment:
This is weird.  Why did the old implementation ignore the sc_len, sc_family, and sc_sysaddr fields?

Reply:
Dunno, it's possible this code is from when that struct was still defined in `nix` and only had these two fields as they were all that was required. Regardless, I think the `libc` implementation is now the correct one.